### PR TITLE
[MANOPD-82392] - add check_iaas tasks for check pakages and repos

### PR DESCRIPTION
**Pull request by Imadzuma**
### Description
During check_iaas we don't check packages and their repositories.

Fixes # (issue)

### Solution
Added two tasks:
* `software.packages.repositories` for check repositories availability;
* `software.packages.availability` for check packages availability;
 

### Test Cases
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**TestCase 1**

Test Configuration:

- Hardware: 
- OS: 
- Inventory: 

Steps:

1. Run `check_iaas`

Results:

| Before | After |
| ------ | ------ |
| No tasks for packages | Two new tasks |

### Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Integration CI passed
- [ ] Unit tests. If Yes list of new/changed tests with brief description
- [ ] There is no merge conflicts

#### Unit tests
Indicate new or changed unit tests and what they do, if any.

### CODE REVIEW

File: kubemarine/procedures/check_iaas.py [hunk 1]
```diff
@@ -314,12 +315,158 @@ def check_access_to_thirdparties(cluster: KubernetesCluster):
...
+
+            if not resolv_conf_actual:
+                warnings.append(f"resolv.conf is not installed for node {node['connect_to']}: "
+                                f"Package repositories can be unavailable. You can install resolv.conf using task "
+                                f"`install --tasks prepare.dns.resolv_conf`")
+
+            # Check with script
+            python_executable = cluster.context['nodes'][node['connect_to']]['python']['executable']
+            for repository_url in repository_urls:
+                res = node['connection'].run("%s %s %s %s" % (python_executable, random_temp_path, repository_url,
```
Comment (ilia1243) [inline kubemarine/procedures/check_iaas.py:414]: It is possible though challenging to use RemoteExecutor to check availability of repositories and packages, and check installed resolv.conf and repositories, for few nodes in parallel.

File: kubemarine/procedures/check_iaas.py [hunk 2]
```diff
@@ -314,12 +315,158 @@ def check_access_to_thirdparties(cluster: KubernetesCluster):
...
+        # Check if package repos are actual
+        package_repos_are_actual = check_package_repositories(node['connection'])
+        if not package_repos_are_actual:
+            warnings.append(f"Package repositories are not installed for {node['connect_to']}: "
+                            f"Packages can be unavailable. You can install it using tasks "
+                            f"`install --tasks prepare.dns.resolv_conf,prepare.package_manager.configure`")
+
+        packages_to_check = cluster.inventory["services"]["packages"].get("install", {}).get('include', []).copy()
+        for association_name, association in cluster.inventory["services"]["packages"].get("associations", {})\
+                .get(node['connection'].get_nodes_os(), {}).items():
```
Comment (ilia1243) [inline kubemarine/procedures/check_iaas.py:447]: Not all associations are in general required. For example, audit is not managed by Kubemarine for rhel OS family (though, probably still need to check it), haproxy/keepalived are not required for some non-HA schemes.

File: kubemarine/procedures/check_iaas.py [hunk 3]
```diff
@@ -314,12 +315,158 @@ def check_access_to_thirdparties(cluster: KubernetesCluster):
...
+                                                 (predefined_repos_file, predefined_repos_file, random_repos_conf_path),
+                                                 warn=True).is_any_failed()
+        # Remove temp .repo file
+        group.sudo("rm %s" % random_repos_conf_path)
+
+    return package_repos_are_actual
+
+
+def check_access_to_package_repositories(cluster: KubernetesCluster):
+    detect_preinstalled_python(cluster)
```
[the remark this answers is not in the document]
Comment (Imadzuma) [reply kubemarine/procedures/check_iaas.py]: `wirh Case` was moved to beginning

File: kubemarine/procedures/check_iaas.py [hunk 4]
```diff
@@ -314,12 +315,158 @@ def check_access_to_thirdparties(cluster: KubernetesCluster):
...
+
+    with TestCase(cluster.context['testsuite'], '013', 'Software', 'Package Repositories') as tc:
+        # Collect repository urls
+        # TODO: think about better parsing
+        repository_urls = set()
+        repositories = cluster.inventory['services']['packages']['package_manager'].get("repositories")
+        cluster_os = cluster.get_os_family()
+        if cluster_os == 'debian':
+            # For debian
+            for repo in repositories:
```
[the remark this answers is not in the document]
Comment (Imadzuma) [reply kubemarine/procedures/check_iaas.py:395]: String was supported

File: kubemarine/procedures/check_iaas.py (hunk 1 above)
Comment (Imadzuma) [reply kubemarine/procedures/check_iaas.py]: Commands were wrapped to RemoteExecutor

File: kubemarine/procedures/check_iaas.py (hunk 2 above)
Comment (Imadzuma) [reply kubemarine/procedures/check_iaas.py]: Added package associations filtration for nodes, where they should be installed

**Review by ilia1243 — APPROVED** [review]

**Review by koryaga — APPROVED** [review]

**Pull request merged**